webpack/assets/javascripts/react_app/components/bookmarks/form/form.test.js

```diff
@@ -34,108 +27,116 @@ describe('bookmark form', () => {
     const { wrapper } = setup();
 
     expect(wrapper.find('BookmarkForm').props().initialValues).toEqual({
-      publik: true,
+      public: true,
       query: '',
+      name: '',
     });
   });
   it('should render the entire form', () => {
     const { wrapper } = setup();
 
     expect(toJson(wrapper)).toMatchSnapshot();
   });
-  it('should convert params for submittion correctly', () => {
-    const spy = jest.spyOn(FormActions, 'submitForm');
-    const { wrapper } = setup();
+  it('should allow creating a private bookmark', async () => {
+    const { wrapper, props } = setup();
 
     wrapper
       .find('input[name="name"]')
-      .simulate('change', { target: { value: 'Joe' } });
+      .simulate('change', { target: { name: 'name', value: 'Joe' } });
     wrapper
       .find('textarea[name="query"]')
-      .simulate('change', { target: { value: 'search' } });
+      .simulate('change', { target: { name: 'query', value: 'search' } });
     wrapper
-      .find('input[name="publik"]')
-      .simulate('change', { target: { value: true } });
+      .find('input[name="public"]')
+      .simulate('change', { target: { name: 'public', value: false } });
 
     wrapper.find('form').simulate('submit');
+
     expect(wrapper.find('.spinner')).toHaveLength(1);
-    expect(spy).toHaveBeenCalledWith({
-      item: 'Bookmark',
-      url: '/api/bookmarks',
-      values: {
-        controller: 'hosts',
-        name: 'Joe',
-        public: true,
-        query: 'search',
-      },
+    await wait(() => {
```

Comment:
You might be able to use the `IntegrationTestHelper.flushAllPromises();` instead:
https://sharvit.github.io/react-redux-test-utils/class/src/IntegrationTestHelper.js~IntegrationTestHelper.html#static-method-flushAllPromises